src/main/java/org/cascadebot/cascadebot/permissions/objects/Result.java

```diff
@@ -0,0 +1,88 @@
+package org.cascadebot.cascadebot.permissions.objects;
+
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+
+public class Result {
```

Comment:
Do we really need this class? As far as i can tell it doesn't provide any info that we don't already have access to when checking permissions in the first place, besides for ResultCause and we can just return that.